src/main/java/com/uci/inbound/utils/XMsgProcessingUtil.java

```diff
@@ -353,7 +353,7 @@ private void sendEventToOutboundKafka(XMessage xmsg) {
     private Mono<XMessageDAO> getLatestXMessage(String userID, XMessage.MessageState messageState) {
         LocalDateTime yesterday = LocalDateTime.now().minusDays(1L);
         return xMsgRepo
-                .findAllByFromIdAndTimestampAfter(userID, yesterday)
+                .findFirstByUserIdInAndFromIdInAndMessageStateInAndTimestampAfterOrderByTimestampDesc(List.of(BotUtil.adminUserId, userID), List.of(BotUtil.adminUserId, userID), List.of(messageState.name()), yesterday)
                 .doOnError(genericError(String.format("Unable to find previous Message for userID %s", userID)))
                 .collectList()
                 .map(xMessageDAOS -> {
```

Comment:
Can we remove this function altogether and use the implementation [here](https://github.com/samagra-comms/inbound/blob/a6ead7c809dfeb3d44e79e05d7177031a0358f3a/src/main/java/com/uci/inbound/utils/XMsgProcessingUtil.java#L508C1-L508C1)?